pkg/controllers/management/multiclusterapp/interval.go

```diff
@@ -0,0 +1,61 @@
+package multiclusterapp
+
+import (
+	"context"
+	"github.com/rancher/rancher/pkg/namespace"
+	"github.com/rancher/rancher/pkg/ticker"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/sirupsen/logrus"
+	"sync"
+	"time"
+)
+
+type IntervalData struct {
+	interval   int
+	cancelFunc context.CancelFunc
+}
+
+var mcAppTickerData map[string]*IntervalData
+var mcAppDataLock = sync.Mutex{}
+
+func storeContext(cctx context.Context, mcapp *v3.MultiClusterApp, mcApps v3.MultiClusterAppInterface) {
+	if mcapp.Spec.UpgradeStrategy.RollingUpdate == nil {
+		return
+	}
+	set := mcapp.Spec.UpgradeStrategy.RollingUpdate.Interval
+	if set == 0 {
+		return
+	}
+	defer mcAppDataLock.Unlock()
+	mcAppDataLock.Lock()
+	if data, ok := mcAppTickerData[mcapp.Name]; ok {
+		if data.interval != set {
+			data.interval = set
+		}
+		return
+	}
+	ctx, cancel := context.WithCancel(cctx)
+	go startTicker(ctx, set, mcApps, mcapp.Name)
```

Comment:
based on input from @kinarashah we have to cancel and start the ticker if the interval changes